lib/srv/ctx.go

```diff
@@ -739,13 +747,11 @@ func getPAMConfig(c *ServerContext) (*PAMConfig, error) {
 	environment["TELEPORT_LOGIN"] = c.Identity.Login
 	environment["TELEPORT_ROLES"] = strings.Join(roleNames, " ")
 	if localPAMConfig.Environment != nil {
-		user, err := c.srv.GetAccessPoint().GetUser(c.Identity.TeleportUser, false)
+		traits, err := services.ExtractTraitsFromCert(c.Identity.Certificate)
```

Comment:
@rosstimothy @russjones Actually, looking at `CreateIdentityContext`, I think it may be fine as it is. It basically fills the `identity.Certificate` field with user certificate so extracting traits there would be the same as extracting them here (what is being done here). It might make sense to run a quick test (even if using just some print statements) but I think it's ok.

Reply:
Spun up a root and leaf cluster locally and verified that the traits were the same when connecting to nodes in the root and leaf clusters. Connected to the leaf cluster node directly through the leaf cluster and through the root cluster.